lib/jsdom/living/xhr/XMLHttpRequest-impl.js

```diff
@@ -83,945 +81,881 @@ const corsSafeResponseHeaders = new Set([
       }
     }
+  }
 
-    setRequestHeader(header, value) {
-      const flag = this[xhrSymbols.flag];
-      const properties = this[xhrSymbols.properties];
+  setRequestHeader(header, value) {
+    const { flag, properties } = this;
 
-      if (arguments.length !== 2) {
-        throw new TypeError("2 arguments required for setRequestHeader");
-      }
-      header = conversions.ByteString(header);
-      value = conversions.ByteString(value);
+    if (this.readyState !== XMLHttpRequestImpl.OPENED || properties.send) {
+      throw DOMException.create(this._globalObject, ["The object is in an invalid state.", "InvalidStateError"]);
+    }
 
-      if (this.readyState !== XMLHttpRequest.OPENED || properties.send) {
-        throw DOMException.create(this._globalObject, ["The object is in an invalid state.", "InvalidStateError"]);
-      }
+    value = normalizeHeaderValue(value);
 
-      value = normalizeHeaderValue(value);
+    if (!tokenRegexp.test(header) || !fieldValueRegexp.test(value)) {
+      throw DOMException.create(this._globalObject, [
+        "The string did not match the expected pattern.",
+        "SyntaxError"
+      ]);
+    }
 
-      if (!tokenRegexp.test(header) || !fieldValueRegexp.test(value)) {
-        throw DOMException.create(this._globalObject, [
-          "The string did not match the expected pattern.",
-          "SyntaxError"
-        ]);
-      }
+    const lcHeader = header.toLowerCase();
 
-      const lcHeader = header.toLowerCase();
+    if (forbiddenRequestHeaders.has(lcHeader) || lcHeader.startsWith("sec-") || lcHeader.startsWith("proxy-")) {
+      return;
+    }
 
-      if (forbiddenRequestHeaders.has(lcHeader) || lcHeader.startsWith("sec-") || lcHeader.startsWith("proxy-")) {
+    const keys = Object.keys(flag.requestHeaders);
+    let n = keys.length;
+    while (n--) {
+      const key = keys[n];
+      if (key.toLowerCase() === lcHeader) {
+        flag.requestHeaders[key] += ", " + value;
         return;
       }
-
-      const keys = Object.keys(flag.requestHeaders);
-      let n = keys.length;
-      while (n--) {
-        const key = keys[n];
-        if (key.toLowerCase() === lcHeader) {
-          flag.requestHeaders[key] += ", " + value;
-          return;
-        }
-      }
-      flag.requestHeaders[header] = value;
-    }
-
-    get _ownerDocument() {
-      return idlUtils.implForWrapper(window.document);
     }
+    flag.requestHeaders[header] = value;
   }
+}
 
-  Object.defineProperty(XMLHttpRequest.prototype, Symbol.toStringTag, {
-    value: "XMLHttpRequest",
-    writable: false,
-    enumerable: false,
-    configurable: true
-  });
-
-  setupForSimpleEventAccessors(XMLHttpRequest.prototype, ["readystatechange"]);
+setupForSimpleEventAccessors(XMLHttpRequestImpl.prototype, ["readystatechange"]);
 
-  addConstants(XMLHttpRequest, {
-    UNSENT: 0,
-    OPENED: 1,
-    HEADERS_RECEIVED: 2,
-    LOADING: 3,
-    DONE: 4
-  });
+addConstants(XMLHttpRequestImpl, {
```

Comment:
```suggestion
Object.assign(XMLHttpRequestImpl, {
```

We should eventually get rid of `addConstants`, so maybe it'd be good to start here. (Since webidl2js will now take care of the property descriptors.)